components/status_dropdown/status_dropdown.jsx

```diff
@@ -105,6 +120,87 @@ export default class StatusDropdown extends React.PureComponent {
             </FormattedMessage>
         );
     }
+    handleClearStatus = (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+        this.props.actions.unsetCustomStatus();
+    };
+
+    onToggle = (open) => this.props.actions.setStatusDropdown(open);
+
+    renderCustomStatus = () => {
+        if (!this.props.isCustomStatusEnabled) {
+            return null;
+        }
+        const customStatus = this.props.customStatus;
+        const isStatusSet = customStatus && (customStatus.text || customStatus.emoji);
+        const customStatusText = isStatusSet ? customStatus.text : localizeMessage('status_dropdown.set_custom', 'Set a Custom Status');
+        const customStatusEmoji = isStatusSet ?
+            (
+                <span className='d-flex'>
+                    <CustomStatusEmoji
+                        showTooltip={false}
+                        emojiStyle={{marginLeft: 0}}
+                    />
+                </span>
+            ) : (
+                <EmojiIcon className={'custom-status-emoji'}/>
+            );
+
+        const clearButton = isStatusSet &&
+            (
+                <span
+                    className='custom_status__clear'
+                >
+                    <OverlayTrigger
+                        delayShow={Constants.OVERLAY_TIME_DELAY}
+                        placement='top'
+                        overlay={
+                            <Tooltip id='clear-custom-status'>
+                                <FormattedMessage
+                                    id='status_dropdown.custom_status.tooltip_clear'
+                                    defaultMessage='Clear'
+                                />
+                            </Tooltip>
+                        }
+                    >
+                        <button
+                            className='style--none input-clear-x'
+                            onClick={this.handleClearStatus}
+                        >
+                            <i className='icon icon-close-circle'/>
+                        </button>
+                    </OverlayTrigger>
+                </span>
+            );
+
+        const pulsatingDot = !isStatusSet && this.props.showCustomStatusPulsatingDot && (
+            <span className='pulsating_dot'/>
+        );
+
+        return (
+            <Menu.Group>
+                <Menu.ItemToggleModalRedux
+                    text={''}
+                    modalId={ModalIdentifiers.CUSTOM_STATUS}
+                    dialogType={CustomStatusModal}
+                    dialogProps={{userId: this.props.userId}}
+                    className='custom_status__row'
+                    id={'status-menu-custom-status'}
+                    sibling={clearButton}
```

Comment:
Is the sibling something we actually need to pass into the menu item as a prop or could we just pass it in as an actual sibling in a fragment here? This seems like an odd pattern to me, although I wouldn't be surprised if it's just another issue with our menu components.

Reply:
@hmhealey We really do need to pass the sibling here to keep the HTML semantic.

If you refer to earlier comments by @deanwhillier and @michelengelen, you'll see that we had earlier implemented it by using a method similar to what you proposed.

But, in the interest of using correct tags in the generated HTML (buttons for both clickable elements and li for the list item), we had to implement it this way.

Any advice on how we could implement this in a better way?

Reply:
Nothing easy comes to mind. Perhaps we could do a custom menu item component so we could have more control over the split menu item, but I'm not sure if that's even really possible with the current component like I mentioned. Let's leave this as-is, and we could always revisit it later when we're rewriting the menu components.